File: lib/classes/book.dart
class Book {
  final String title;
  final String description;
  final String format;
  final String imageURL;
  final String creator;
  final int amount;
  final String author;
  Book({
    required this.title,
    required this.description,
    required this.format,
    required this.imageURL,
    required this.creator,
    required this.amount,
    required this.author,
  });

  factory Book.fromJson(Map<String, dynamic> book) {
    return Book(
      title: book['title'],
      description: book['description'],
      format: book['format'],
      imageURL: book['imageURL'],
      creator: book['creator'],
      amount: book['amount'],
      author: book['author'],
    );
  }
}

File: models/book.js
const { Int32 } = require('bson');
const mongoose = require('mongoose');

const Schema = mongoose.Schema;


const bookSchema = new Schema({
    title: {
        type: String,
        required: true,
    },
    author: {
        type: String,
        required: true,
    },
    format: {
        type: String,
        required: true,
    },
    imageURL : {
        type: String,
        required: true,
    },
    rating: {
        type: Number,
    },
    description: {
        type: String,
        required: true,
    },
    creator: {
        type: Schema.Types.ObjectId,
    },
    amount: {
        type: Number,
        required: true,
    },
});

module.exports = mongoose.model('Book', bookSchema);